coding-challenges.js

```diff
@@ -39,22 +52,37 @@ var nums2 = [109, 5, 9, -59, 8, 24]
 
 // --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
 
-
-
-
-
 var testString1 = "albatross"
 // Expected output: "aLbAtRoSs"
 var testString2 = "jabberwocky"
 // Expected output: "jAbBeRwOcKy"
 
-
+const alternateCapitale = (string) => {
+	let newString = []
+	for(let i = 0; i < string.length; i++){
+		if(i%2 != 0)
+			newString.push(string[i].toUpperCase())
+		else
+			newString.push(string[i])
+
+	}
+	return newString.join("")
+}
+console.log(alternateCapitale(testString1))
+console.log(alternateCapitale(testString2))
 // --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
 
-
-
-
-
 var arr1 = [3, 7, 10, 5, 4, 3, 3]
 var arr2 = [7, 8, 2, 3, 1, 5, 4]
 // Expected output: [3, 7, 10, 5, 4, 8, 2, 1]
+
+const uniqueValues = (arr1, arr2) => {
+	let newSet = new Set()
+	for(let i = 0; i < arr1.length; i++)
+		newSet.add(arr1[i])
+	for(let i = 0; i < arr2.length; i++)
+		newSet.add(arr2[i])
+
+	return Array.from(newSet)
+}
+console.log(uniqueValues(arr1, arr2))
```

Comment:
good use of Set. obv a very efficient solution. just be prepared for an interviewer to ask you to write out the long way. i always suggest considering an optimized solution but for the sake of practice, i would put a "naive" solution (meaning, less efficient or possibly your initial approach), and an "optimized" solution in your assessment. this practice will have a real-world application as well as it is _extremely_ helpful/impressive in an interview setting. showing prospective employers how you approach a problem and how you go about refactoring is very valuable.